### PR TITLE
refactor: rename Catalog to ApplicationCatalog

### PR DESCRIPTION
**Pull request by axel7083**
### What does this PR do?

Rename the Catalog interface to ApplicationCatalog as we want to have later a dedicated ModelCatalog. See https://github.com/projectatomic/ai-studio/issues/447 for details

### What issues does this PR fix or reference?

Fixes https://github.com/projectatomic/ai-studio/issues/488

### How to test this PR?

- [x] existing tests ensure no regression

### CODE REVIEW

**Comment by axel7083** [comment]
> Reference

Sorry I included a previous commit that should not have been here. Fixed

File: packages/frontend/src/lib/Navigation.svelte [hunk 1]
```diff
@@ -17,7 +17,7 @@ export let meta: TinroRouteMeta;
     </div>
   </div>
   <div class="h-full overflow-hidden hover:overflow-y-auto" style="margin-bottom:auto">
-    <SettingsNavItem title="Recipe Catalog" href="/recipes" bind:meta="{meta}" />
+    <SettingsNavItem title="Recipe ApplicationCatalog" href="/recipes" bind:meta="{meta}" />
```
Comment (axel7083) [inline packages/frontend/src/lib/Navigation.svelte:20]:
```suggestion
    <SettingsNavItem title="Recipe Catalog" href="/recipes" bind:meta="{meta}" />
```

File: docs/proposals/state-management.md [hunk 2]
```diff
@@ -9,17 +9,17 @@ at initial time.
 
 The pages of the front-end subscribe to the store to get the value of the state in a reactive manner.
 
-## Catalog
+## ApplicationCatalog
```
Comment (axel7083) [inline docs/proposals/state-management.md:12]:
```suggestion
## Catalog
```

File: docs/proposals/state-management.md [hunk 3]
```diff
@@ -9,17 +9,17 @@ at initial time.
...
 
 The catalog is persisted as a file in the user's filesystem. The backend reads the file at startup,
 and watches the file for changes. The backend updates the state as soon as changes it detects changes.
 
-The front-end uses a `readable` store, which waits for changes on the Catalog state 
+The front-end uses a `readable` store, which waits for changes on the ApplicationCatalog state
```
Comment (axel7083) [inline docs/proposals/state-management.md:17]:
```suggestion
The front-end uses a `readable` store, which waits for changes on the Catalog state 
```

File: docs/proposals/state-management.md [hunk 4]
```diff
@@ -9,17 +9,17 @@ at initial time.
...
 and watches the file for changes. The backend updates the state as soon as changes it detects changes.
 
-The front-end uses a `readable` store, which waits for changes on the Catalog state 
+The front-end uses a `readable` store, which waits for changes on the ApplicationCatalog state 
 (using `onMessage('new-catalog-state', data)`),
 and asks for the current state at startup (with `postMessage('ask-catalog-state')`).
 
 The interested pages of the front-end subscribe to the store to get the value
-of the Catalog state in a reactive manner.
+of the ApplicationCatalog state in a reactive manner.
```
Comment (axel7083) [inline docs/proposals/state-management.md:22]:
```suggestion
of the Catalog state in a reactive manner.
```

**Review by lstocchi — APPROVED** [review]
LGTM

**Comment by axel7083** [comment]
cc @jeffmaury

**Review by jeffmaury — APPROVED** [review]
LGTM

**Pull request merged**